modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbTableStorage.java

```diff
@@ -0,0 +1,316 @@
+
+    /** Busy lock to stop synchronously. */
+    final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    private final ScheduledExecutorService scheduledExecutor;
+
+    /** Thread pool to execute after-flush actions. */
+    private final ExecutorService threadPool;
+
+    /** Supplier for the value of delay for scheduled database flush. */
+    private final IntSupplier flushDelaySupplier;
+
+    /**
+     * Constructor.
+     *
+     * @param tableCfg Table configuration.
+     * @param dbPath Database path.
+     * @param scheduledExecutor Scheduled executor.
+     */
+    public TxStateRocksDbTableStorage(
+            TableConfiguration tableCfg,
+            Path dbPath,
+            ScheduledExecutorService scheduledExecutor,
+            ExecutorService threadPool,
+            IntSupplier flushDelaySupplier
+    ) {
+        this.tableCfg = tableCfg;
+        this.dbPath = dbPath;
+        this.scheduledExecutor = scheduledExecutor;
+        this.threadPool = threadPool;
+        this.flushDelaySupplier = flushDelaySupplier;
+    }
+
+    /**
+     * Checks that a passed partition id is within the proper bounds.
+     *
+     * @param partitionId Partition id.
+     */
+    private void checkPartitionId(int partitionId) {
+        if (partitionId < 0 || partitionId >= storages.length()) {
+            throw new IllegalArgumentException(S.toString(
+                "Unable to access partition with id outside of configured range",
+                "table", tableCfg.name().value(), false,
+                "partitionId", partitionId, false,
+                "partitions", storages.length(), false
+            ));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public TxStateStorage getOrCreateTxStateStorage(int partitionId) throws StorageException {
+        checkPartitionId(partitionId);
+
+        TxStateStorage storage = new TxStateRocksDbStorage(
+                db,
+                writeOptions,
+                readOptions,
+                persistedTierReadOptions,
+                partitionId,
+                this
+        );
+
+        storages.set(partitionId, storage);
+
+        return storage;
+    }
+
+    /** {@inheritDoc} */
+    @Override public @Nullable TxStateStorage getTxStateStorage(int partitionId) {
+        return storages.get(partitionId);
+    }
+
+    /** {@inheritDoc} */
+    @Override public CompletableFuture<Void> destroyTxStateStorage(int partitionId) throws StorageException {
+        checkPartitionId(partitionId);
+
+        TxStateStorage storage = storages.get(partitionId);
+
+        if (storage != null) {
+            try {
+                storage.close();
+            } catch (Exception e) {
+                throw new StorageException("Couldn't close the transaction state storage of partition "
+                        + partitionId + ", table " + tableCfg.value().name());
+            }
+
+            storage.destroy();
+        }
+
+        return completedFuture(null);
```

Comment:
Can you please check what MvTableStorage does? I doubt that it returns a completed future.